fadecandy_driver/src/ros/fadecandy_driver_ros.cpp

```diff
@@ -0,0 +1,86 @@
+//
+// Copyright (c) 2021 Eurotec
+//
+
+#include "./fadecandy_driver_ros.h"
+#include <fadecandy_msgs/LEDArray.h>
+
+namespace fadecandy_driver
+{
+FadecandyDriverRos::FadecandyDriverRos()
+{
+  ros::NodeHandle nh;
+
+  timer_ = nh.createTimer(ros::Duration(1.), &FadecandyDriverRos::timerCallback, this);
+
+  diagnostic_updater_.add("Info", this, &FadecandyDriverRos::diagnosticsCallback);
+
+  led_subscriber = nh.subscribe<fadecandy_msgs::LEDArray>("/set_leds", 1, &FadecandyDriverRos::setLedsCallback, this);
+  initialized_ = false;
+}
+
+void FadecandyDriverRos::run()
+{
+  if (fadecandy_device_ != NULL)
+  {
+    release();
+    fadecandy_device_ = NULL;
+  }
+
+  if (!initialized_)
+  {
+    initialized_ = FadecandyDriverRos::intialize();
+    if (initialized_)
+    {
+      ROS_INFO("Connected to Fadecandy device");
+
+      diagnostic_updater_.setHardwareID(serial_number_);
+    }
+    else
+    {
+      throw std::runtime_error("Device not found! Could not write on the driver.");
+    }
+  }
```

Comment:
spin here if you call the method `run`

Reply:
I did it in the beginning  this way but then I changed it to  ros::spinOnce() in the node  because once I put the ros::spin()  here it never reaches it once the run threw and error which means stopping the diagnostic topic of publishing the status of the driver.

Reply:
then your design is wrong, if you want this; you need a timer callback that calls the reconnect and not introduce a separate look. Who is throttling the rate now?